h2/src/test/org/h2/test/unit/TestPgServer.java

```diff
@@ -521,6 +526,44 @@ private void testTextualAndBinaryTypes(boolean binary) throws SQLException {
         }
     }
 
+    private void testBinaryNumeric() throws SQLException {
+        if (!getPgJdbcDriver()) {
+            return;
+        }
+        Server server = createPgServer(
+                "-ifNotExists", "-pgPort", "5535", "-pgDaemon", "-key", "pgserver", "mem:pgserver");
+        supportedBinaryOids.add(1700);
+        try {
+            Properties props = new Properties();
+            props.setProperty("user", "sa");
+            props.setProperty("password", "sa");
+            // force binary
+            props.setProperty("prepareThreshold", "-1");
+
+            Connection conn = DriverManager.getConnection(
+                    "jdbc:postgresql://localhost:5535/pgserver", props);
+            Statement stat = conn.createStatement();
+
+            try (ResultSet rs = stat.executeQuery("SELECT 1E-16383, 1E-16384, 1E+1")) {
+                rs.next();
+                assertEquals(new BigDecimal("1E-16383"), rs.getBigDecimal(1));
+                for (int i = 2; i <= 3; i ++) {
+                    try {
+                        rs.getBigDecimal(i);
+                        fail();
+                    } catch (IllegalArgumentException e) {
+                        // PgJDBC doesn't support scale greater than 16383 or negative scale
+                    }
+                }
+            }
```

Comment:
`SELECT 1E1` works with PostgreSQL/PgJDBC well, `getBigDecimal()` returns 10 as expected. I think you need to normalize scale in your implementation too.

Reply:
Normalized in the latest commit. May still have problem for `1E+131072` or other edge cases. (Write an incorrect value or throw exception? I choose writing the incorrect value because PgJDBC and Npgsql will reject them.)